CHANGELOG.md

```diff
@@ -41,6 +43,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Structuring concrete subclasses no longer requires providing an explicit `type` field
 - `_target(s)` attributes of `Objectives` are now de-/serialized without leading
   underscore to support user-friendly serialization strings
+- Running simulations no longer affects the states of the random number generators
```

Comment:
Hi @AVHopp, @Scienfitz. The change fixes the RNG problem pointed out by @brandon-holt. However, I'm not 100% happy with the implementation since it requires adding yet another function that completely copies the long signature. IMO, the ideal solution would be to just implement a decorator that "adds" the `random_seed` argument to the existing signature of the now called `_simulate_experiment_given_seed` function. Unfortunately, I can't get it properly typed... In principle, there is a mechanism for that using `typing.Concatenate`, but we really want to have it as a keyword argument and not an additional positional argument, and unfortunately this is not (yet) supported and has been rejected in the corresponding [PEP](_simulate_experiment_given_seed).

Do you have any other ideas? If not, we can simply keep it as is. I mean, it does the job after all ...

Reply:
Given the number of open PRs at the moment and since the only issue here is a bit more boilerplate code, I would be fine with keeping this as-is since I also do not see a better option. I could imagine that me might be able to control this better depending on how we want to refactor this subpackage in general (e.g. if we want to have a `Simulator` class or something like that, maybe we can do some stuff?), but for now this is sufficient imo.